### DISCUSSION TITLE
saving feed

### DISCUSSION THREAD

**Discussion by roycw** [bluenviron/gortsplib#584] [Questions]
Hello,

Can i save an RTSP in mp4 chunck of lets say 10 seconds ? i only saw the HLS example and i wonder if that is possible.

Also, any recommedation on how to consum it on the browser.
I know f a few LIBS that support it but wondering if there is a companion for MSE.

Also, as a final question: if there is no way to save continues mp4 and i need to spin ffmpeg for that.. Is there anty reason to use this lib to fetch the stream to begin with or is it better to just span ffmpeg and deal with the RTSP connection from the start.

thanks so much for your help.

**Comment by aler9** [comment]
Hello, gortsplib, as the name implies, has been conceived as a RTSP library, not as a general purpose streaming and media conversion library. Examples are provided just to show how the library can be combined with other code in order to perform recording or data conversion.

If you want to consume a RTSP stream into the browser, i suggest you take a look at MediaMTX, that performs this conversion automatically with no code involved:
https://github.com/bluenviron/mediamtx

> Is there anty reason to use this lib to fetch the stream to begin with or is it better to just span ffmpeg and deal with the RTSP connection from the start.

This depends on the use case. If you are working exclusively in Go, then it's probably better to use the library since it's more performant, furthermore Go's parallelism allows to run thousands of instances of this library without any particular hardware requirement. If on the other hand you need more flexibility and the number of streams is limited, then you can spawn FFmpeg.
